backend/src/app/modules/course/course.service.ts

```
@@ -78,13 +77,17 @@ const getAllCourses = async (payload: any, user: null | JwtPayload) => {
+    let whereTerms: any = {
         instructorId: payload.filters?.instructorId,
         title: {
             contains: payload.filters?.title,
         },
         isDeleted: false,
-        isPublished: (process.env.NODE_ENV !== 'development')
+        isPublished: true
+    }
+
+    if (process.env.NODE_ENV === 'development') {
+        delete whereTerms.isPublished
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)